polynote-server/src/main/scala/polynote/server/repository/FileBasedRepository.scala

```diff
@@ -209,8 +209,11 @@ class FileBasedRepository(
           pending.shutdown *> closed.succeed(()).unit <* ZIO.fail(None),
           _ => ZIO.dieMessage("Unreachable state"),
           chunk => chunk.mapM {
-            case ((update, completerOpt)) =>
-              (writeWAL(update).forkDaemon &> doUpdate(update) >>= notifyCompleter(completerOpt)).asSomeError <& updatesTopic.publish(update)
```

Comment:
The problem was that here, it was doing `publish(update)` but that doesn't have the real new global version in it yet. The new global version gets computed by `doUpdate`.